pkg/apis/certificates/validation/validation.go

```diff
@@ -572,3 +580,256 @@ func validateTrustBundle(path *field.Path, in string) field.ErrorList {
+	if req.Status.BeginRefreshAt == nil {
+		allErrors = append(allErrors, field.Required(beginRefreshPath, "must be present and at least 10 minutes before the certificate expires"))
+		return allErrors
+	}
+	if req.Status.BeginRefreshAt.Time.Before(req.Status.NotBefore.Time.Add(10 * time.Minute)) {
+		allErrors = append(allErrors, field.Invalid(beginRefreshPath, req.Status.BeginRefreshAt.Time, "must be at least 10 minutes after status.NotBefore"))
+		return allErrors
+	}
+	if req.Status.BeginRefreshAt.Time.After(req.Status.NotAfter.Time.Add(-10 * time.Minute)) {
+		allErrors = append(allErrors, field.Invalid(beginRefreshPath, req.Status.BeginRefreshAt.Time, "must be at least 10 minutes before status.notAfter"))
+		return allErrors
+	}
+
+	// Check the remainder of the certificates in the chain, if any.  We cannot
+	// easily verify the chain, because the Golang X.509 libraries are wisely
+	// written to prevent us from doing stupid things like verifying a partial
+	// chain, but we can at least check that they are valid certificates.
+	for {
+		var nextBlock *pem.Block
+		nextBlock, rest = pem.Decode(rest)
+		if nextBlock == nil {
+			break
+		}
+
+		if nextBlock.Type != "CERTIFICATE" {
+			allErrors = append(allErrors, field.Invalid(certChainPath, req.Status.CertificateChain, "issued certificate chain must consist entirely of CERTFICATE PEM blocks"))
+			return allErrors
+		}
+
+		_, err := x509.ParseCertificate(nextBlock.Bytes)
+		if err != nil {
+			allErrors = append(allErrors, field.Invalid(certChainPath, req.Status.CertificateChain, "intermediate certificate does not parse as valid X.509"))
+			return allErrors
+		}
+	}
+
+	// TODO(KEP-4317): Tests
+
+	return allErrors
+}
+
+func timeNear(a, b time.Time, skew time.Duration) bool {
+	return a.After(b.Add(-skew)) && a.Before(b.Add(skew))
+}
+
+func hashBytes(in []byte) []byte {
+	out := sha256.Sum256(in)
+	return out[:]
+}
+
+var (
+	pkixPath         = field.NewPath("spec", "pkixPublicKey")
+	popPath          = field.NewPath("spec", "proofOfPossession")
+	certChainPath    = field.NewPath("status", "certificateChain")
+	issuedAtPath     = field.NewPath("status", "issuedAt")
+	notBeforePath    = field.NewPath("status", "notBefore")
+	notAfterPath     = field.NewPath("status", "notAfter")
+	beginRefreshPath = field.NewPath("status", "beginRefreshAt")
+)
+
+// ValidatePodCertificateRequestUpdate runs all update validation checks on an
+// update.
+func ValidatePodCertificateRequestUpdate(newReq, oldReq *certificates.PodCertificateRequest, optsIn ValidatePodCertificateRequestOptions) field.ErrorList {
+	opts := optsIn
+	if bytes.Equal(newReq.Spec.PKIXPublicKey, oldReq.Spec.PKIXPublicKey) && bytes.Equal(newReq.Spec.ProofOfPossession, oldReq.Spec.ProofOfPossession) && newReq.Status.CertificateChain == oldReq.Status.CertificateChain {
+		opts.SkipCryptoValidation = true
+	}
+
+	var allErrors field.ErrorList
+	allErrors = append(allErrors, ValidatePodCertificateRequest(newReq, opts)...)
```

Comment:
This doesn't delegate because the update is immutable.

Reply:
Done.